src/components/page/Page.tsx

```diff
@@ -93,36 +94,58 @@ const Page = (props: { sketch: Sketch }) => {
         newHeight = (newWidth / config.width) * config.height;
       }
     }
+    const canvas = getCanvas();
     canvas.canvas.style.height = newHeight + 'px';
     canvas.canvas.style.width = newWidth + 'px';
   };
 
-  const draw = () => {
+  const restart = () => {
+    const sketchProps = getSketchProps();
+    sketch.reset(sketchProps);
+    sketch.init(sketchProps);
+    draw();
+    loopState.restart();
+  };
+
+  const getCanvas = () => {
     // Grab our canvas
     const pageCanvas = document.getElementById(
       'sketch-canvas',
     ) as HTMLCanvasElement;
-    canvas = new Canvas(pageCanvas);
-    // updateSketchProps();
+    return new Canvas(pageCanvas);
+  };

```

Comment:
Canvas element gets replaced frequently on any props update (hot reloading), so it's very convenient to always get the element from the page and regenerate the (stateless) Canvas object when we need a canvas.